### ISSUE TITLE
Support multiple document stores

### ISSUE THREAD

**Issue by ThomWright** [pwcremin/RNSync#5]
Currently `rnsync` is a singleton which only supports a single store. We need to be able to use multiple stores.

There is an implementation in [this fork](https://github.com/calvium/RNSync/tree/add-indexes) which I could port over, but I reckon we could make a better API.

The above fork produces an API which requires a store name in each method:

```js
import rnsync from "rnsync"

rnsync.init("http://databaseUrl", "storeName", callback)

rnsync.replicatePull("storeName", callback)
```

I'd prefer an API where I could initialise and get a reference to multiple stores:

```js
import rnsync from "rnsync"

rnsync.init("http://databaseUrl", (store) => {
  store.replicatePull(callback)
})
```

Thoughts?

**Comment by ThomWright** [comment]
Context: we're currently using the fork I linked above, but I'd like to port over the changes and stop using the fork.

Some other things I think we'll need:

- `createIndexes()` - apparently on a background thread on Android
- `getAttachmentsFor(documentId)`
- `deleteStore()`

[Changes in the fork](https://github.com/pwcremin/RNSync/compare/a5e80...calvium:add-indexes)

**Comment by pwcremin** [comment]
@ThomWright I stole some of your code and created 4.1.0

```
import {RNSync} from 'rnsync'

let store1 = new RNSync(cloudantUrl, 'testdb1')
await store1.init()
            .catch( e => console.log('ERROR failed to init rnSync: ' + e))
store1.create({x:10})

let store2 = new RNSync(cloudantUrl, 'testdb2')
await store2.init()
            .catch( e => console.log('ERROR failed to init rnSync: ' + e))
store2.create({x:10})
```

etc.

Please try it out and let me know if it meets your needs.

I left RNSyncWrapper as the default export in index.js for backwards compat

**Comment by ThomWright** [comment]
Hi @pwcremin 

We're actually working on a [PR for this](https://github.com/candide-eu/RNSync/tree/port-calvium-fork).

I prefer the API we've created (see below) over the one above. It has the advantage that it doesn't give access to a store object until the database has been initialised, which means it doesn't allow you to call e.g. `store.create(...)` on an invalid store.

```js
import rnsync from "rnsync"

rnsync.init(cloudantUrl, "databaseName", (store) => {
  // ...
})

// or

const store = await rnsync.init(cloudantUrl, "databaseName")
```

**Comment by pwcremin** [comment]
I agree, I just didn't want to do another major revision so soon (breaks backwards compat).  We can definitely do one though.  Submit the PR when it is ready

**Comment by pwcremin** [comment]
@ThomWright I am not sure if you noticed but I made you a collaborator for this project.  Update as you need to get the package in shape for your team

**Comment by ThomWright** [comment]
Thanks @pwcremin, we're still testing the changes but it should be ready soon.
